hypothesis-python/src/hypothesis/provisional.py

```diff
@@ -51,17 +69,46 @@ def url_encode(s):
 
 
 @st.defines_strategy_with_reusable_values
-def domains():
-    """A strategy for :rfc:`1035` fully qualified domain names."""
-    atoms = st.text(
-        string.ascii_letters + "0123456789-", min_size=1, max_size=63
-    ).filter(lambda s: "-" not in s[0] + s[-1])
+def domains(
+    max_length=255,  # type: int
+    max_element_length=63,  # type: int
+):
+    # type: (...) -> SearchStrategy[Text]
+    """A strategy for :rfc:`1035` fully qualified domain names.
+
+    The max_length is 255 in accordance with :rfc:`1035#section-2.3.4`
+    The maximum_element_length is 63 in accordance with :rfc:`1035#section-2.3.4`
```

Comment:
nit:

```suggestion
    The max_element_length is 63 in accordance with :rfc:`1035#section-2.3.4`
```

To match the name in the parameter list.

Reply:
Thanks for spotting that, proofreading is hard. 😄